### PR TITLE
more clean up

### PR DESCRIPTION
**Pull request by anadeem717**

### CODE REVIEW

File: BankAccount/Account.cs [hunk 1]
```diff
@@ -1,8 +1,8 @@
 
 public class BankAccount {
 
-    public string fName { get; private set; }
-    public string lName { get; private set; }
+    public string firstName { get; private set; }
```
Comment (yj7o5) [inline BankAccount/Account.cs]: the industry in general has adopted various casing when naming variables camel case, pascal case, etc https://curc.readthedocs.io/en/latest/programming/coding-best-practices.html 

in c# and dotnet in general people use pascal casing for naming public object properties, so in this case `FirstName`, `LastName`, etc

File: BankAccount/Account.cs [hunk 2]
```diff
@@ -41,7 +41,7 @@ public class BankAccount {
 
     // displays account info for the user's account
    public void ViewAccountInfo() {
-    Console.WriteLine($"\nACCOUNT INFO FOR {this.fName} {this.lName}:");
+    Console.WriteLine($"\nACCOUNT INFO FOR {this.firstName} {this.lastName}:");
```
Comment (yj7o5) [inline BankAccount/Account.cs]: sometimes its a bit of a code smell to keep using `Console.WriteLine` again and again, check this for what it means about code smell https://www.wikiwand.com/en/Code_smell. There's a better class that can utilize in c# that will make your code look more succinct such as this `StringBuilder` https://learn.microsoft.com/en-us/dotnet/api/system.text.stringbuilder?view=net-8.0 which can make your code look something like this:

```
var sb = new StringBuilder();
sb.AppendFormat("Account INFO FOR {0} {1}", this.FirstName, this.LastName);
...
...
Console.WriteLine(sb.ToString());
```
this way you have only 1 `Console.WriteLine()`

Comment (yj7o5) [reply BankAccount/Account.cs]: you don't really have to use `Console.WriteLine` everywhere just where you see when you have several consecutive `Console.WriteLine` statements `StringBuilder` can make it less verbose

File: BankAccount/Program.cs [hunk 3]
```diff
@@ -12,7 +12,7 @@ static void Main(string[] args)
         List<BankAccount> Accounts = new List<BankAccount>();
```
Comment (yj7o5) [inline BankAccount/Program.cs]: this is generally more of a personal preference but C# has this `var` keyword that infers the type info without you have to explicitly type it all the time, it makes code look a bit more cleaner
```
var Accounts = new List<BankAccount>();
``` 

https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/implicitly-typed-local-variables

also inside a function/method we don't name the local variables with uppercase should be ~Account~ account

File: BankAccountTests/BankAccountTests.cs [hunk 4]
```diff
@@ -17,7 +17,7 @@ public void PasswordStrengthCheck(string password)
     [InlineData("a!Hshj")]
     public void PasswordStrengthCheck2(string password)
```
Comment (yj7o5) [inline BankAccountTests/BankAccountTests.cs]: why is there another test method? you're checking the same `Program.CheckPassword` in both the test functions the only difference is a different test data so just put the `[InlineData("a!Hshj")]` in the above password check method as well. there's a general principal in programming and i.e. K.I.S.S. (Keep It Super Simple) https://dev.to/kwereutosu/the-k-i-s-s-principle-in-programming-1jfg

File: BankAccountTests/BankAccountTests.cs [hunk 5]
```diff
@@ -1,14 +1,14 @@
 
 namespace BankAccountTests;
-public class UnitTest1
+public class BankAccountTests
 {
     [Theory]
     [InlineData("HelloWorld123!")]
     [InlineData("helloworldH$09")]
     [InlineData("aaaAaaaaa123455678?")]
```
Comment (yj7o5) [inline BankAccountTests/BankAccountTests.cs]: also in your test cases you should check for both true positives and true negative, right now it only checks for true positive. a better way to test your function would be:
```
InlineData("HelloWorld123!", true)
...
InlineData("", false)
public void TestCheckPassword(string password, bool expectedValue)
...
    Assert.Equal(Program.CheckPassword(password), expectedValue)
...
```

this way you make sure that you are testing _all_ the test cases, when it correctly determines the right and wrong passwords.

File: BankAccountTests/BankAccountTests.cs [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
     public void PasswordStrengthCheck(string password)
```
Comment (yj7o5) [inline BankAccountTests/BankAccountTests.cs]: also read up on this, you should follow the best practices for when naming and constructing your test cases:

https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#naming-your-tests

so in this case you could have named your test method as:

`Check_Password_Returns_Correct_Strength` or something like that

File: BankAccountTests/BankAccountTests.cs [hunk 7]
```diff
@@ -26,8 +26,8 @@ public void AccountCheck()
     {
         BankAccount account = new BankAccount("John", "Doe", "jdoe", "HelloWorld123!", 100);
 
-        Assert.Equal("John", account.fName);
-        Assert.Equal("Doe", account.lName);
+        Assert.Equal("John", account.firstName);
+        Assert.Equal("Doe", account.lastName);
         Assert.Equal("jdoe", account.username);
         Assert.Equal("HelloWorld123!", account.password);
         Assert.Equal(100, account.balance);
```
Comment (yj7o5) [inline BankAccountTests/BankAccountTests.cs]: In your `BalanceCheck` test method it only tests for the happy path, you should also check for if your current balance is 100 and you withdraw 110 it wouldn't do anything because of insufficient balance and it should still be 100.

**Review by yj7o5 — CHANGES_REQUESTED** [review]

File: BankAccountTests/BankAccountTests.cs [hunk 8]
```diff
@@ -3,54 +3,56 @@ namespace BankAccountTests;
...
-        Assert.Equal(149.6, account.balance);
+    [Fact]
+    public void Withdraw_DecreasesBalance()
+    {
+        BankAccount account = new BankAccount("John", "Doe", "jdoe", "HelloWorld123!", 100);
 
-        account.Deposit(100);
-        Assert.Equal(249.6, account.balance);
+        account.Withdraw(50.50);
+        Assert.Equal(49.50, account.Balance);
```
Comment (yj7o5) [inline BankAccountTests/BankAccountTests.cs:47]: its still missing a test case where if you withdraw more amount than the existing balances then the account balance doesn't go into negative

File: BankAccountTests/BankAccountTests.cs [hunk 9]
```diff
@@ -3,54 +3,56 @@ namespace BankAccountTests;
...
 
-        account.Withdraw(100);
-        Assert.Equal(149.6, account.balance);
+    [Fact]
+    public void Withdraw_InsufficientBalance_DoesNotChangeBalance()
+    {
+        BankAccount account = new BankAccount("John", "Doe", "jdoe", "HelloWorld123!", 100);
 
+        account.Withdraw(110);
+        Assert.Equal(100, account.Balance);
```
Comment (yj7o5) [inline BankAccountTests/BankAccountTests.cs]: Now that you have a more robust test case for your program. I would request one last change here. What you are doing in your program it is good behavior that it does not incorrectly withdraw amount more than the existing balance. However, simply just logging out that you "insufficient balance" isn't enough. For example, If I am making use of this class `account.Withdraw(110);` and I see that account.Balance hasn't change there should be a more proactive response in my program if I could get to know before hand what is wrong with my program. This is where **Exceptions** come into play. I would recommend reading into exceptions. https://learn.microsoft.com/en-us/dotnet/standard/exceptions

Your task here is to modify the program such that it throws a certain type of exception when the condition of insufficient balance hits in your program. You should create a `InsufficientBalanceException` and then throw that inside the program when you try to `Withdraw` balance. Then in your test case you should assert that it throws that specific type of exception when you try to with draw more. 

https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting.assert.throwsexception?view=visualstudiosdk-2022

**Review by yj7o5 — CHANGES_REQUESTED** [review]
Just one last change otherwise all the rest of the changes looks perfect. Great job.
